### PR TITLE
[Core][cli][usability] ray stop prints errors during graceful shutdown 

### LINKED ISSUE #25518: [cli][usability] ray stop prints errors during graceful shutdown
### What happened + What you expected to happen

Sometimes when Ray is gracefully stopped with `ray stop` some of the helper processes still exit with non-zero exit codes. Ideally we should always exit 0 to avoid false alarms

### Versions / Dependencies

Master

### Reproduction script

https://discuss.ray.io/t/ray-on-slurm-shutdown-throws-errors/5939

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
This is to address false alarms on subprocesses exiting when killed by `ray stop` with SIGTERM.

## What has been changed?
1. Added signal handlers for some of the subprocesses:
  - dashboard (head)
  - log monitor
  - ray client server 

2. Changed the `--block` semantics and prompt messages. 

## Related issue number
Closes #25518

### CLI test:
**Before** 
```
ray start --head --block
# In another shell
ray stop 
```
image

------- 

**Now**
image

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/log_monitor.py [hunk 1]
```diff
@@ -503,6 +512,16 @@ def is_proc_alive(pid):
         args.logs_dir, gcs_pubsub.GcsPublisher(address=args.gcs_address), is_proc_alive
     )
 
+    # Set up handlers for graceful exit on SIGINT and SIGTERM
+    def graceful_exit(signum: int, _frame):
```
Comment (rkooo567) [inline python/ray/_private/log_monitor.py:516]: Btw, just curious, but how does this work? Isn't the main thread already occupied by the loop? 

IIRC, python runs the interrupt on a main thread.

**Review by rkooo567 — COMMENTED** [review]
Can you add tests? Ideally,
1. Test the actual issue
2. Test if non sigterm/sigint is received, it prints them
3. Test when components are shutdown with sigterm/sigint, they exit with 0

File: python/ray/_private/log_monitor.py (hunk 1 above)
Comment (rickyyx) [reply python/ray/_private/log_monitor.py:516]: Yeah, but apparently there is another abstraction between python and the underlying OS according to this: https://docs.python.org/3/library/signal.html#execution-of-python-signal-handlers

Not sure the exact code paths, but I believe it is OS -> some python code -> runs signal handler in main thread.

**Comment by iycheng** [comment]
Overall it looks good to  me.

File: python/ray/tests/test_cli.py [hunk 2]
```diff
@@ -303,6 +327,167 @@ def test_ray_start_hook(configure_lang, monkeypatch, tmp_path):
...
 
+@pytest.mark.skipif(
+    sys.platform == "darwin"
+    and "travis" in os.environ.get("USER", "")
+    or sys.platform == "win32",
+    reason=(
+        "Mac builds don't provide proper locale support. "
+        "Windows signal handling not compatible"
+    ),
+)
```
Comment (iycheng) [inline python/ray/tests/test_cli.py:338]: We can have two pytest.mark.skipif I think.
```
@pytest.mark.skipif(sys.platform == "win32", reaons = "...")
@pytest.mark.skipif(...)

```

**Review by fishbone — APPROVED** [review]
LG.

**Comment by rickyyx** [comment]
Should be ready to merge. @rkooo567

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**